src/EnergyPlus/UnitarySystem.cc

```diff
@@ -534,11 +519,7 @@ namespace UnitarySystems {
                 this->m_MyFanFlag = false;
             } else {
                 if (this->m_FanExists) {
-                    if (this->m_FanType == DataHVACGlobals::FanType::SystemModel) {
-                        this->m_ActualFanVolFlowRate = state.dataHVACFan->fanObjs[this->m_FanIndex]->designAirVolFlowRate;
-                    } else {
-                        this->m_ActualFanVolFlowRate = Fans::GetFanDesignVolumeFlowRate(state, this->m_FanIndex);
-                    }
+                    this->m_ActualFanVolFlowRate = state.dataFans->fans(this->m_FanIndex)->maxAirFlowRate;
```

Comment:
Does this need to be guarded or is `m_FanExists` enough?

Reply:
It is possible there is a fan object in the UnitarySystem and the fan properties have not been read in yet. This code will continue to execute until a fan is found. So using `if (this->m_FanExists && this->m_FanIndex > 0)` should work. Or, reading fan properties could be moved above the return shown below where this->m_FanExists = true is set.

In getInput there are UnitarySystem properties read in prior to this code to get a minimal amount of information and then more detailed information is read in after this point (after the simulation has read in all other objects).

    // Early calls to ATMixer don't have enough info to pass GetInput. Need to get the data next time through.
    if (sysNum == -1 || !state.dataZoneEquip->ZoneEquipInputsFilled) {
        return;
    }

Reply:
This whole `GetInput` dance needs to change. It's not good.